jerry-core/ecma/operations/ecma-regexp-object.h

```diff
@@ -18,6 +18,33 @@
 
 #ifndef CONFIG_DISABLE_REGEXP_BUILTIN
 
+#ifdef REGEXP_RECURSION_LIMIT
+/**
+ * Decrease the recursion counter and test it.
+ * If the counter reaches the limit of the recursion depth
+ * it will return with a range error.
+ */
+#define REGEXP_RECURSION_COUNTER_DECREASE_AND_TEST() \
+  if (--re_ctx_p->recursion_counter == 0) \
+    { \
+      return ecma_raise_range_error (ECMA_ERR_MSG ("RegExp executor recursion limit is exceeded.")); \
+    }
```

Comment:
This is not following the coding style: align curly braces with the `if`.

Moreover, to make this a proper macro that acts like a statement, wrap it into a `do { ... } while (0)`.